internal/aws/containerinsight/utils.go

```diff
@@ -0,0 +1,230 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package containerinsight
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+)
+
+// aggregate fileds
+func AggregateFields(fields []map[string]interface{}) map[string]float64 {
+	if len(fields) == 0 {
+		return nil
+	}
+
+	result := make(map[string]float64)
+	// Use the first element as the base
+	for k, v := range fields[0] {
+		result[k] = v.(float64)
+	}
+
+	if len(fields) == 1 {
+		return result
+	}
+
+	for i := 1; i < len(fields); i++ {
+		for k, v := range result {
+			if fields[i][k] == nil {
+				continue
+			}
+			result[k] = v + fields[i][k].(float64)
+		}
+	}
+	return result
+}
+
+func IsNode(mType string) bool {
+	return mType == TypeNode || mType == TypeNodeNet || mType == TypeNodeFS || mType == TypeNodeDiskIO
```

Comment:
Using a `switch` might be easier to maintain going forward.  This also applies to the remaining predicates that follow this one.

```suggestion
	switch mtype {
	case TypeNode, TypeNodeNet, TypeNodeFS, TypeNodeDiskIO:
		return true
	}
	return false
```

Reply:
Make sense. I will modify it.